conans/client/cmake.py

```diff
@@ -128,14 +135,22 @@ def flags(self):
                 flags.extend(["-DCONAN_CXX_FLAGS=-m32",
                               "-DCONAN_SHARED_LINKER_FLAGS=-m32",
                               "-DCONAN_C_FLAGS=-m32"])
+            elif op_system == "Windows" and host_platform != "Windows":
+                flags.extend(["-DCMAKE_C_COMPILER=i686-w64-mingw32-gcc",
+                              "-DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++"])
             elif op_system == "Macos":
                 flags.append("-DCMAKE_OSX_ARCHITECTURES=i386")
-
+        elif arch == "x86_64":
+            if op_system == "Windows" and host_platform != "Windows":
+                flags.extend(["-DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc",
+                              "-DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++"])
         try:
             libcxx = self._settings.compiler.libcxx
             flags.append('-DCONAN_LIBCXX="%s"' % libcxx)
         except:
             pass
+
+        print("Flags: " + " ".join(flags))
```

Comment:
sorry it was a typo error, I've refactored that for clang as well. The point is to identify if we are compiling for a target platform and the hosting platform where conan runs is equal to the target or not.